docs/content/guides/developer/app-examples/reviews-rating.mdx

```diff
@@ -0,0 +1,451 @@
+A [dynamic field](../../../concepts/dynamic-fields.mdx) stores the list of services that are registered to a dashboard.
+
+:::info
+
+See [Shared versus Owned Objects](../sui-101/shared-owned.mdx) for more information on the differences between object types.
+
+:::
+
+### review.move
+
+This module defines the `Review` struct.
+
+```move
+/// Represents a review of a service
+public struct Review has key, store {
+    id: UID,
+    owner: address,
+    service_id: ID,
+    content: String,
+    // intrinsic score
+    len: u64,
+    // extrinsic score
+    votes: u64,
+    time_issued: u64,
+    // proof of experience
+    has_poe: bool,
+    total_score: u64,
+    overall_rate: u8,
+}
+
+/// Updates the total score of a review
+fun update_total_score(rev: &mut Review) {
+    rev.total_score = rev.calculate_total_score();
+}
+
+/// Calculates the total score of a review
+fun calculate_total_score(rev: &Review): u64 {
+    let mut intrinsic_score: u64 = rev.len;
+    intrinsic_score = math::min(intrinsic_score, 150);
+    let extrinsic_score: u64 = 10 * rev.votes;
+    // VM = either 1.0 or 2.0 (if user has proof of experience)
+    let vm: u64 = if (rev.has_poe) { 2 } else { 1 };
+    (intrinsic_score + extrinsic_score) * vm
+}
+```
+
+In addition to the content of a review, all the elements that are required to compute total score are stored in a `Review` object.
+
+A `Review` is a [shared object](../../../concepts/object-ownership/shared.mdx), so anyone can cast a vote on a review and update its `total_score` field.
+After `total_score` is updated, the [`update_top_reviews`](#casting-votes) function can be called to update the `top_reviews` field of the `Service` object.
+
+### service.move
+
+This module defines the `Service` struct that service owners manage.
+
+```move
+const MAX_REVIEWERS_TO_REWARD: u64 = 10;
+
+/// Represents a service
+public struct Service has key, store {
+    id: UID,
+    reward_pool: Balance<SUI>,
+    reward: u64,
+    top_reviews: vector<ID>,
+    reviews: ObjectTable<ID, Review>,
+    overall_rate: u64,
+    name: String
+}
+```
+
+#### Reward distribution
+
+The same amount is rewarded to top reviewers, and the reward is distributed to 10 participants at most.
+The pool of `SUI` tokens to be distributed to reviewers is stored in the `reward_pool` field, and the amount of `SUI` tokens awarded to each participant is configured in `reward` field.
+
+#### Storage for reviews
+
+Because anyone can submit a review for a service, `Service` is defined as a shared object. All the reviews are stored in the `reviews` field, which has [ObjectTable](../../../concepts/sui-move-concepts/collections.mdx#object_table)`<ID, Review>` type. The `reviews` are stored as children of the shared object, but they are still accessible by their `ID`.
+In other words, anyone can go to a transaction explorer and find a review object by its `object_id` but they won't be able to use a review as an input to a transaction by its `object_id`.
```

Comment:
nit: `object_id` is not an identifier we've introduced in code anywhere, so prefer leaving it unformatted.

```suggestion
In other words, anyone can go to a transaction explorer and find a review object by its object ID but they won't be able to use a review as an input to a transaction by its object ID.
```